### ISSUE TITLE
Clean up index.rst for Sphinx docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7044]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7044|
|Type|enhancement|
|Created|2014-03-20 16:47:56Z|
|Branch|https://github.com/twisted/twisted/tree/clean-up-indexrst-7044-2|

The index.rsts in the Twisted documentation have a lot of split lines, blank lines, and weird source formatting. They should be brought up to standard, so they're easier to add things to at a later date.

<details><summary>Searchable metadata</summary>

```
trac-id__7044 7044
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_clean_up_indexrst_7044_2 branches/clean-up-indexrst-7044-2
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1395334076000000 1395334076000000
changetime__1423499882842110 1423499882842110
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

This patch does

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

This patch does it.

It is based off [#6926](https://github.com/twisted/twisted/issues/6926) - moving everything down one level and editing `conf.py` is from that - this is the rest of the changes.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Please be more specific than this in ticket descriptions.  "messy" could mean a lot of different things.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

> docs/web/examples/simple.rtl

This change seems unrelated.  Changing the filename breaks it, too.  If the file needs to be renamed (why does it need to be renamed?) please file a separate ticket for doing so in a way that won't break the example.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban commented|
|-|-|

Rather than ```+needs_sphinx = '1.2'``` maybe this branch should consider the work from [#6767](https://github.com/twisted/twisted/issues/6767) 
and not put sphinx requirement in conf.py

Having version requirements in multiple places could make it harder to keep information in sync.

If the version is included in conf.py maybe we need a ticket to remove it (or read it from here) once twisted dev/build environment is upgraded to something more robust: ex virtualenv + pip requirements

-----

This branch looks like a move for directories from docs/projects/* to /docs/ 

As commented by exarkun I thing that the rename of simple.rtl needs a comment as I don't think it was accidental.

------

It also contain indentation changes and empty line removal for files.

I checked the changes and they look good.

-----

I ran a fresh sphinx build on my local computer on this branch (using sphinx 1.2.2) and apart from warning reported and fixed in [#6970](https://github.com/twisted/twisted/issues/6970) no other warning or errors were generated.

```
$ sphinx-build --version
Sphinx (sphinx-build) 1.2.2

$ sphinx-build -b html -d _build/doctrees  -n -q -a -E . _build/html
/home/adi/twisted/docs/web/howto/web-in-60/access-logging.rst:30: WARNING: Pygments lexer name u'shell' is not known

Build finished. The HTML pages are in _build/html.
```

Thanks!

**Comment by twisted-trac** [comment]
|<img alt="bramwelt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|bramwelt commented|
|-|-|

Replying to [adiroiban](#note_7):
> Rather than ```+needs_sphinx = '1.2'``` maybe this branch should consider the work from [#6767](https://github.com/twisted/twisted/issues/6767) 
> and not put sphinx requirement in conf.py
> 
> Having version requirements in multiple places could make it harder to keep information in sync.
> 
> If the version is included in conf.py maybe we need a ticket to remove it (or read it from here) once twisted dev/build environment is upgraded to something more robust: ex virtualenv + pip requirements

Listing ``` need_sphinx = '1.2' ``` is fine with me in _conf.py_. This does not install Sphinx, or set any installed version of Sphinx to this version. It merely restricts _sphinx-builder_ from building docs without a version of Sphinx >= 1.2. So isn't necessarily a version requirement.

> ```
> $ sphinx-build -b html -d _build/doctrees  -n -q -a -E . _build/html
> /home/adi/twisted/docs/web/howto/web-in-60/access-logging.rst:30: WARNING: Pygments lexer name u'shell' is not known
> ```

Running sphinx-build on Trunk produced this error as well, so I don't believe it is a byproduct of this change.

> As commented by exarkun I thing that the rename of simple.rtl needs a comment as I don't think it was accidental. 

I don't think this file needs to be renamed, just that the link to the file needs to be fixed.

----

hawkowl,

Overall this is a great change, it's just lacking some context/comments as other have pointed out.

My only suggestion is to switch the note at the top of _docs/lore/index.rst_ to actually use the ``` .. note:: ``` or even ``` .. attention:: ``` sphinx-doc [admonition](http://docutils.sourceforge.net/docs/ref/rst/directives.html#note).

Thanks for cleaning up these index files!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_10"></a>|@glyph set owner to @hawkowl|
|-|-|

The comments from bramwelt, adiroiban, and exarkun seem to constitute a fairly thorough review.  Looks like this needs a bit more work.  Please address those comments and resubmit.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl commented|
|-|-|

(In [43893]) Branching to clean-up-indexrst-7044-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_12"></a>|@hawkowl removed owner|
|-|-|

Merged forward (manually!) and everything is good.

I realised why it was an .rtl and fixed the download link, so it will download now.

Spinning buildbots, putting up for review.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_14"></a>|@hynek commented|
|-|-|

AFAICT, everything has been addressed and the result LGTM.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)|@hynek set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_16"></a>|@hawkowl set status to `closed`|
|-|-|

(In [43911]) Merge clean-up-indexrst-7044-2: Clean up Sphinx index.rst files

Author: hawkowl
Reviewers: exarkun, adiroiban, bramwelt, hynek
Fixes: [#7044](https://github.com/twisted/twisted/issues/7044)
